From fd4459d69f9bb5dc2009771208daacb841e5873b Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@chainguard.dev>
Date: Thu, 2 May 2024 18:16:47 +0100
Subject: [PATCH 1/4] go: cmd/go: always emit ldflags version information

Even when -trimpath is active, emit full ldflags in the version
information ELF note. Vulnerability scanners typically parse ldflags
field to detect main package version, thus binaries that are built
with -trimpath are currently actively evading vulnerability scanners.

Fixes: https://github.com/golang/go/issues/63432

Signed-off-by: Dimitri John Ledkov <dimitri.ledkov@chainguard.dev>
Change-Id: Ifd0e1171fca82182113012a4432ea3b1c4706fb2
---
 src/cmd/go/internal/load/pkg.go | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/cmd/go/internal/load/pkg.go b/src/cmd/go/internal/load/pkg.go
index 2985cff67c..e949410129 100644
--- a/src/cmd/go/internal/load/pkg.go
+++ b/src/cmd/go/internal/load/pkg.go
@@ -2475,9 +2475,7 @@ func (p *Package) setBuildInfo(ctx context.Context, f *modfetch.Fetcher, autoVCS
 		// determine whether they may refer to system paths. If we do that, we can
 		// redact only those paths from the recorded -ldflags setting and still
 		// record the system-independent parts of the flags.
-		if !cfg.BuildTrimpath {
-			appendSetting("-ldflags", ldflags)
-		}
+		appendSetting("-ldflags", ldflags)
 	}
 	if cfg.BuildCover {
 		appendSetting("-cover", "true")
-- 
2.53.0

